### ISSUE TITLE
optimizer error?

### ISSUE THREAD

**Issue by WERush** [renll/RSAM#1]
In this line, https://github.com/renll/RSAM/blob/master/RSAM2layerLOC2.py#L230, you define a new optimizer at every batch. Is it right? 
I think that optimizer should be initialized at the start only once, and so it can change the parameters in the model and hyperparameters in itself with the epoch goes by.

**Comment by renll** [comment]
I don't think so. You can check the Pytorch online docs and find that the SGD optimizer does not "memorize" anything between each optimization step. The point that I define a new optimizer every batch is that it allows me to have a dynamic change of the learning rate.

**Comment by WERush** [comment]
Can you give me an example code that defines optimizer at every batch?

If you want to make adaptive learning rate, a method is like as following:
```
def adjust_learning_rate(optimizer, epoch):
    lr = lr * 0.95 # or changing lr with respect to epoch
    for param_group in optimizer.param_groups:
        param_group['lr'] = lr
```

**Comment by renll** [comment]
You can check the Pytorch example Word level Language Modeling using LSTM RNNs and prove that my optimizer is the same as the version that without gradient clips.

**Comment by renll** [comment]
Anyway, I admit that it is a more natural way to change the learning rate as you proposed, and I may revise my code as soon as I am free from deadlines:) Thanks a lot for pointing out this.

**Comment by WERush** [comment]
Thanks, i will check this example code.

I am just curious about this manner of optimizer definition. No offense. Thanks!
